metricbeat/helper/elastic/elastic.go

```diff
@@ -158,8 +158,31 @@ func NewModule(base *mb.BaseModule, xpackEnabledMetricsets []string, logger *log
 		return nil, errors.Wrapf(err, "could not unpack configuration for module %v", moduleName)
 	}
 
-	// These metricsets are exactly the ones required if xpack.enabled == true
-	raw["metricsets"] = xpackEnabledMetricsets
+	// Ensure all required metricsets are enabled when xpack.enabled == true
```

Comment:
@miltonhultgren We should extend this later to add a flag that allows us to configure metricsets individually again even if xpack is enabled. This ensures we can have cluster stats with a longer cycle.

Reply:
I think @joshdover has a PR that lays some foundation for that.

Reply:
Yes that PR should help without breaking the current functionality. Allows for enabling new metricsets without allow users to disable any required ones in xpack mode